packages/core/src/external/carequality/pd/validating-pd.ts

```diff
@@ -0,0 +1,185 @@
+  if (!firstName) {
+    throw new InternalError("Given name is not defined");
+  }
+  const lastName = patient.name?.[0]?.family;
+  if (!lastName) {
+    throw new InternalError("Family name is not defined");
+  }
+  const birthDate = patient.birthDate;
+  if (!birthDate) {
+    throw new InternalError("Birth date is not defined");
+  }
+
+  const genderAtBirth =
+    patient.gender === "male" ? "M" : patient.gender === "female" ? "F" : undefined;
+  if (!genderAtBirth) {
+    throw new LivingSubjectAdministrativeGenderRequestedError("Gender at Birth is not defined");
+  }
+
+  const addresses = (patient.address ?? []).map((addr: FHIRAddress) => {
+    const addressLine1 = addr.line ? addr.line.join(" ") : "";
+    const city = addr.city || "";
+    const state = addr.state ? getStateEnum(addr.state) : undefined;
+    const zip = addr.postalCode || "";
+    const country = addr.country || "USA";
+
+    if (!addressLine1) {
+      throw new PatientAddressRequestedError("Address Line 1 is not defined");
+    }
+    if (!city) {
+      throw new PatientAddressRequestedError("City is not defined");
+    }
+    if (!state) {
+      throw new PatientAddressRequestedError("State is not defined");
+    }
+    if (!zip) {
+      throw new PatientAddressRequestedError("Zip is not defined");
+    }
+
+    const newAddress: Address = {
+      addressLine1,
+      city,
+      state,
+      zip,
+      country,
+    };
+    return newAddress;
+  });
+
+  const contacts = (patient.telecom ?? []).map((tel: ContactPoint) => {
+    const contact: Contact = {};
+    if (tel.system && isContactType(tel.system)) {
+      contact[tel.system] = tel.value;
+    }
+    return contact;
+  });
+
+  const personalIdentifiers = (patient.identifier ?? [])
+    .map((identifier: Identifier) => {
+      const system = identifier.system;
+      const value = identifier.value;
+      if (system && value && stateMappings[system]) {
+        const state = stateMappings[system];
+        if (state) {
+          const personalIdentifier: PersonalIdentifier = {
+            type: "driversLicense",
+            value: value,
+            state: state,
+          };
+          return personalIdentifier;
+        }
+      }
+      return undefined;
+    })
+    .filter((item): item is PersonalIdentifier => item !== undefined);
+
+  // TODO: this is a nessecary consequence of using Patient here, since we are taking a demographic type
+  // and converting it to the full patient type
+  const convertedPatient: Patient = {
+    cxId: "",
+    eTag: "",
+    createdAt: new Date(),
+    updatedAt: new Date(),
+    facilityIds: [],
+    id: "",
+    data: {
+      firstName: firstName,
+      lastName: lastName,
+      dob: birthDate,
+      genderAtBirth: genderAtBirth,
+      address: addresses,
+      contact: contacts,
+      personalIdentifiers: personalIdentifiers,
+    },
+  };
+  return convertedPatient;
```

Comment:
This is the other time where its required to fill a patient with empty values, because there is no actual Patient object yet, just incoming demographics